src/vs/workbench/contrib/positronKeybindings/browser/positronKeybindings.contribution.ts

```diff
@@ -0,0 +1,273 @@
+import * as nls from '../../../../nls.js';
+import { positronConfigurationNodeBase } from '../../../services/languageRuntime/common/languageRuntime.js';
+import { IConfigurationService } from '../../../../platform/configuration/common/configuration.js';
+import { KeybindingsRegistry, KeybindingWeight } from '../../../../platform/keybinding/common/keybindingsRegistry.js';
+import { KeyCode, KeyMod } from '../../../../base/common/keyCodes.js';
+import { EditorContextKeys } from '../../../../editor/common/editorContextKeys.js';
+import { ContextKeyExpr } from '../../../../platform/contextkey/common/contextkey.js';
+
+const configurationRegistry = Registry.as<IConfigurationRegistry>(
+	Extensions.Configuration
+);
+configurationRegistry.registerConfiguration({
+	...positronConfigurationNodeBase,
+	properties: {
+		'workbench.keybindings.rstudioKeybindings': {
+			scope: ConfigurationScope.MACHINE,
+			type: 'boolean',
+			default: false,
+			description: nls.localize('keybindings.rstudioKeybindings', "Enable RStudio keybindings (requires restart)"),
+		},
+	}
+});
+
+class PositronKeybindingsContribution extends Disposable {
+
+	static readonly ID = 'workbench.contrib.positronKeybindings';
+
+	private readonly _registrations: DisposableStore = new DisposableStore();
+
+	constructor(
+		@IConfigurationService private readonly _configurationService: IConfigurationService
+	) {
+		super();
+
+		this._register(this._registrations);
+
+		// If the configuration setting is enabled, register the RStudio key mappings
+		const rstudioKeybindingsEnabled =
+			this._configurationService.getValue('workbench.keybindings.rstudioKeybindings');
+		if (rstudioKeybindingsEnabled) {
+			this.registerRStudioKeybindings();
+		}
+
+		// Listen for changes to the configuration setting.
+		//
+		// In practice it appears that updating the registry doesn't take effect
+		// until the next startup, so unfortunately this doesn't enable us to
+		// dynamically toggle the setting within a single Positron session.
+		this._register(
+			this._configurationService.onDidChangeConfiguration((e) => {
+				if (e.affectsConfiguration('workbench.keybindings.rstudioKeybindings')) {
+					// Handle the change in configuration
+					const rstudioKeybindingsEnabled =
+						this._configurationService.getValue('workbench.keybindings.rstudioKeybindings');
+					if (rstudioKeybindingsEnabled) {
+						// Register the key mappings
+						this.registerRStudioKeybindings();
+					} else {
+						// Unregister the key mappings by clearing the registrations
+						this._registrations.clear();
+					}
+				}
+			})
+		);
+	}
+
+	/**
+	 * Registers the RStudio key mappings with the keybinding registry.
+	 */
+	private registerRStudioKeybindings() {
+		// Create new R file
+		this._registrations.add(KeybindingsRegistry.registerKeybindingRule({
+			id: 'r.createNewFile',
+			weight: KeybindingWeight.WorkbenchContrib,
+			primary: KeyMod.CtrlCmd | KeyCode.KeyN
```

Comment:
```suggestion
			primary: KeyMod.CtrlCmd |  KeyMod.Shift | KeyCode.KeyN
```
I think you forgot the shift key here:
https://github.com/posit-dev/positron/blob/64988dfaeaa5de9219cb777569c2e8631aefbe0e/extensions/positron-rstudio-keymap/package.json#L53-L58